Bug: no validation and confirmation on comments

Dante291 opened the pull request:
**What kind of change does this PR introduce?**

This PR introduces an enhancement by implementing validation and confirmation mechanisms for comments.

**Issue Number:**

Fixes #2277 

**Did you add tests for your changes?**

Yes

**Snapshots/Videos:**

[validation.webm](https://github.com/PalisadoesFoundation/talawa/assets/103507835/a5fa34a6-d6f4-47ff-81f3-a51a597c2d99)


**If relevant, did you update the documentation?**

No

**Does this PR introduce a breaking change?**

No

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa/blob/master/CONTRIBUTING.md)?**

Yes

Cioppolo14 commented:
Please fix the failing tests.

Dante291 commented:
@palisadoes @Cioppolo14 please assign reviewers.

Cioppolo14 commented:
@literalEval @anwersayeed Can you review this PR?

Dante291 commented:
@noman2002 @literalEval please review this

literalEval commented on lib/views/after_auth_screens/feed/individual_post.dart:
```diff
@@ -59,14 +70,32 @@ class _IndividualPostViewState extends State<IndividualPostView> {
+                    )
+                  : null,
+
+              ///check if button is enabled when comment is valid
```
Use `//` for comments.

literalEval commented on lib/views/after_auth_screens/feed/individual_post.dart:
```diff
@@ -104,6 +133,14 @@ class _IndividualPostViewState extends State<IndividualPostView> {
+/// * `Padding`: padding for the text
 Padding buildPadding(BuildContext context, String text) {
   return Padding(
     padding: const EdgeInsets.symmetric(vertical: 8.0),
```
Use dynamic sizing instead.

literalEval commented on lib/views/after_auth_screens/feed/individual_post.dart:
```diff
@@ -121,6 +158,7 @@ class IndividualPageLikeSection extends StatelessWidget {
     required this.usersLiked,
   });
 
+  /// List of user who liked the post.
```
*users

literalEval commented on lib/views/after_auth_screens/feed/individual_post.dart:
```diff
@@ -154,7 +192,11 @@ class IndividualPostCommentSection extends StatelessWidget {
     required this.postID,
   });
+
+  /// LIst of comments on a post.
```
*List

literalEval commented on lib/views/after_auth_screens/feed/individual_post.dart:
```diff
@@ -228,7 +271,13 @@ class CommentTemplate extends StatelessWidget {
+/// * `user`: User who liked the post
+///
+/// **returns**:
+/// * `Widget`: returns the circle avatar of the likedBy user who liked the post
```
Don't use words like returns, takes etc. It is obvious. Moreover, make the comment multi-line if the single line goes too wide.

Dante291 commented:
@literalEval made requested changes

literalEval reviewed (changes requested):
There are some guidelines which you should try on enforcing yourself, like:

1. Writing better comments.
2. Adding trailing commas and refactoring.
3. Adding proper spacing on comments and docs.

Dante291 commented:
@literalEval 

1. I have revisited and have added more meaningful docs in the code to provide better clarity.
2.I have added trailing commas wherever necessary for proper readability of code.
3. Added proper spacing on comments and docs.

If, after these updates, there are still areas that require improvement, please feel free to specify the errors or provide additional guidance. Your feedback is highly valued, and I'm committed to delivering code that meets the highest standards.

noman2002 reviewed (approved):
LGTM
